CDK: migration steps for partners (#326)
0xPolygon/polygon-docs: docs/cdk/how-to/migrate/7-8.md

> This document shows Polygon partners how to migrate an isolated  CDK stack from fork 7 to fork 8.

Author: ```suggestion
This document shows Polygon partners how to migrate an isolated CDK stack.
```

## Process to upgrade forks for isolated CDK chains

In order to avoid reorgs and other undesirable scenarios, all L2 transactions must be verified before upgrading a fork. Verification means that all batches are closed, sequenced, and verified on L1.

> Follow the steps to verify all batches for upgrading from fork 7 to 8.

Author: ```suggestion
Follow the steps to verify all batches for upgrading.
```

    4. Check sequencer halted when reaching batch `X+1` (this is obvious in the logs).

    5. Wait until all pending batches are virtualized (X): 

        `SELECT batch_num FROM state.virtual_batch ORDER BY batch_num DESC LIMIT 1;` → X

    6. Wait until the aggregator has aggregated proofs for all the batches:

        1. `SELECT batch_num FROM state.verified_batch ORDER BY batch_num DESC LIMIT 1;` → Y (if Y == X) you can skip next steps until `3. Prepare (**do not apply**) new versions according to the version matrix`

        2. `SELECT batch_num, batch_num_final FROM state.proof WHERE NOT generating AND batch_num = Y AND batch_num_final = X` wait until this query returns a row, remove `WHERE` conditions to get a sense of progress.

    7. Edit node config to force the aggregator into sending the already aggregated proof ASAP: 

        `Aggregator.VerifyProofInterval = "5m”`. Then restart aggregator.

    8. Wait until the proof is settled on-chain: 

        `SELECT batch_num FROM state.verified_batch ORDER BY batch_num DESC LIMIT 1;` → X

3. Prepare (**do not apply**) new versions according to the version matrix.

4. Stop all services (node, prover/executor, bridge).

> 

Author: ```suggestion
```

## Update software

1. Start synchronizer's new version.

> 2. Wait until synchornizer receives a fork id 8 event (check table `state.fork_id`).

Author: ```suggestion
2. Wait until synchornizer receives a fork id event (check table `state.fork_id`).
```

This document shows Polygon partners how to migrate an isolated  CDK stack from fork 7 to fork 8.

> 

Author: ```suggestion
```

> ## Process to upgrade from fork 7 to 8 for isolated CDK chains

Author: ```suggestion
## Process to upgrade forks for isolated CDK chains
```

    4. Check sequencer halted when reaching batch `X+1` (this is obvious in the logs).

    5. Wait until all pending batches are virtualized (X): 

        `SELECT batch_num FROM state.virtual_batch ORDER BY batch_num DESC LIMIT 1;` → X

    6. Wait until the aggregator has aggregated proofs for all the batches:

        1. `SELECT batch_num FROM state.verified_batch ORDER BY batch_num DESC LIMIT 1;` → Y (if Y == X) you can skip next steps until `3. Prepare (**do not apply**) new versions according to the version matrix`

        2. `SELECT batch_num, batch_num_final FROM state.proof WHERE NOT generating AND batch_num = Y AND batch_num_final = X` wait until this query returns a row, remove `WHERE` conditions to get a sense of progress.

    7. Edit node config to force the aggregator into sending the already aggregated proof ASAP: 

        `Aggregator.VerifyProofInterval = "5m”`. Then restart aggregator.

    8. Wait until the proof is settled on-chain: 

        `SELECT batch_num FROM state.verified_batch ORDER BY batch_num DESC LIMIT 1;` → X

3. Prepare (**do not apply**) new versions according to the version matrix.

4. Stop all services (node, prover/executor, bridge).

> 5. Ask Polygon to send the L1 tx, and wait until confirmation.

Author: ```suggestion
```

> This document shows Polygon partners how to migrate a CDK stack from fork 7 to fork 8.

Reviewer: ```suggestion
This document shows Polygon partners how to migrate an isolated  CDK stack from fork 7 to fork 8.
```

!!! tip

>     This process is done collaboratively with Polygon the CDK team.

Reviewer: please delete this sentence

> !!! tip
>     This process is done collaboratively with Polygon the CDK team.

Author: ```suggestion
```

    4. Check sequencer halted when reaching batch `X+1` (this is obvious in the logs).

    5. Wait until all pending batches are virtualized (X): 

        `SELECT batch_num FROM state.virtual_batch ORDER BY batch_num DESC LIMIT 1;` → X

    6. Wait until the aggregator has aggregated proofs for all the batches:

        1. `SELECT batch_num FROM state.verified_batch ORDER BY batch_num DESC LIMIT 1;` → Y (if Y == X) you can skip next steps until `3. Prepare (**do not apply**) new versions according to the version matrix`

        2. `SELECT batch_num, batch_num_final FROM state.proof WHERE NOT generating AND batch_num = Y AND batch_num_final = X` wait until this query returns a row, remove `WHERE` conditions to get a sense of progress.

    7. Edit node config to force the aggregator into sending the already aggregated proof ASAP: 

        `Aggregator.VerifyProofInterval = "5m”`. Then restart aggregator.

    8. Wait until the proof is settled on-chain: 

        `SELECT batch_num FROM state.verified_batch ORDER BY batch_num DESC LIMIT 1;` → X

3. Prepare (**do not apply**) new versions according to the version matrix.

4. Stop all services (node, prover/executor, bridge).

> 5. Ask Polygon to send the L1 tx, and wait until confirmation.

Reviewer: Remove this sentence

This document shows Polygon partners how to migrate a CDK stack from fork 7 to fork 8.

!!! tip
    This process is done collaboratively with Polygon the CDK team.

> ## Process to upgrade from fork 7 to 8

Reviewer: ```suggestion
## Process to upgrade from fork 7 to 8 for isolated CDK chains
```

> 1. CDK owners make an upgrade request to Polygon.
> 
> 2. Polygon schedules the update transaction, and informs the owners of the date and time that the timelock delay expires, after which point the upgrade can go ahead.
> 
> 3. Polygon and CDK owners agree upon a time window for the upgrade.
> 
> 4. During the agreed time window, CDK operators ensure that all the batches of the system are verified. Follow the [verify all batches](#verify-all-batches) instructions below to do that. 
> 
>     !!! warning
>         The current batch verification procedure relies on stopping the sequencer.
> 
> 5. After successful batch verification, CDK operators inform Polygon that their CDK is ready for the upgrade.
> 
> 6. Polygon executes the scheduled transaction on the timelock, making the upgrade effective.
> 
> 7. Polygon informs the CDK operators of the transaction confirmation. (It is also possible to check Etherscan for confirmation of the same.)
> 
> 8. CDK operators update the software and restart all components. Follow the [update software](#update-software) instructions below.

Author: ```suggestion
```

> 
> ## Verify all batches
> 

Author: ```suggestion
```
